### PR TITLE
Workflow Update: return NotFound error to the pollers of outcome of accepted updates of completed workflows

### PR DESCRIPTION
**Pull request by alexshtin**
## What changed?

Return `NotFound` error to the pollers of outcome of accepted updates of completed workflows.

## Why?

If workflow is completed and update was left in `Accepted` stage it will never go `Completed`, therefore there is no reason to wait for update result.

## How did you test it?

Added unit test.

## Potential risks

No risks.

## Documentation

No.

## Is hotfix candidate?

No.

### CODE REVIEW

**Review by stephanos — APPROVED** [review]

File: service/history/workflow/update/registry.go [hunk 1]
```diff
@@ -188,12 +189,18 @@ func NewRegistry(
 				withInstrumentation(&r.instrumentation),
 			)
 		} else if acc := updInfo.GetAcceptance(); acc != nil {
-			r.updates[updID] = newAccepted(
+			uc := newAccepted(
```
Comment (dandavison) [inline service/history/workflow/update/registry.go:192]: nit: why "uc"?
```suggestion
			u := newAccepted(
```

**Review by dandavison — APPROVED** [review]
Looks like an improvement. I agree that we should add a test for this.

File: service/history/workflow/update/registry.go (hunk 1 above)
Comment (alexshtin) [reply service/history/workflow/update/registry.go:192]: "updateAccepted" :-) Let it be just `u`.

**Comment by alexshtin** [comment]
Added unit test.

**Pull request merged**